### PR TITLE
Tensorflow GPU on cpu machine/container

### LINKED ISSUE #37689: spurious error:  failed call to cuInit: UNKNOWN ERROR
ubuntu 18.04, CUDA 10.1, TF 2.1, python 3.6.5

on a machine *without* a GPU, the following code emits "error:  failed call to cuInit: UNKNOWN ERROR".  of course that should fail!  you shouldn't have even tried.  printing that requires extra logic when parsing log files to check whether jobs succeeded.

```
Python 3.6.5 (default, Apr  1 2018, 05:46:30) 
[GCC 7.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import tensorflow as tf
2020-03-18 08:16:50.221103: W tensorflow/stream_executor/platform/default/dso_loader.cc:55] Could not load dynamic library 'libnvinfer.so.6'; dlerror: libnvinfer.so.6: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /usr/local/cuda/lib64:/.singularity.d/libs
2020-03-18 08:16:50.221260: W tensorflow/stream_executor/platform/default/dso_loader.cc:55] Could not load dynamic library 'libnvinfer_plugin.so.6'; dlerror: libnvinfer_plugin.so.6: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /usr/local/cuda/lib64:/.singularity.d/libs
2020-03-18 08:16:50.221275: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:30] Cannot dlopen some TensorRT libraries. If you would like to use Nvidia GPU with TensorRT, please make sure the missing libraries mentioned above are installed properly.
>>> msg = tf.constant('Hello, TensorFlow!')
2020-03-18 08:16:55.455153: W tensorflow/stream_executor/platform/default/dso_loader.cc:55] Could not load dynamic library 'libcuda.so.1'; dlerror: libcuda.so.1: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /usr/local/cuda/lib64:/.singularity.d/libs
2020-03-18 08:16:55.455184: E tensorflow/stream_executor/cuda/cuda_driver.cc:351] failed call to cuInit: UNKNOWN ERROR (303)
2020-03-18 08:16:55.455223: I tensorflow/stream_executor/cuda/cuda_diagnostics.cc:156] kernel driver does not appear to be running on this host (h10u02.int.janelia.org): /proc/driver/nvidia/version does not exist
2020-03-18 08:16:55.455545: I tensorflow/core/platform/cpu_feature_guard.cc:142] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA
2020-03-18 08:16:55.468119: I tensorflow/core/platform/profile_utils/cpu_utils.cc:94] CPU Frequency: 2099935000 Hz
2020-03-18 08:16:55.469505: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x49b71e0 initialized for platform Host (this does not guarantee that XLA will be used). Devices:
2020-03-18 08:16:55.469532: I tensorflow/compiler/xla/service/service.cc:176]   StreamExecutor device (0): Host, Default Version
```

### PR DESCRIPTION
**Pull request by bhack**
Try to fix: https://github.com/tensorflow/tensorflow/issues/37689

Please note that this a blackbox patch cause I am still waiting for a sustainable solution for sparse code contribution at https://github.com/tensorflow/build/issues/5

### CODE REVIEW

**Comment by bhack** [comment]
@chsigg What do you think?

**Comment by bhack** [comment]
@gbaned If @chsigg is busy in these days let me know if you can re-route the review cause I don't know if I will have a free slot in the next days.

**Review by chsigg — APPROVED** [review]

**Comment by bhack** [comment]
@chsigg thanks

File: tensorflow/stream_executor/cuda/cuda_driver.cc [hunk 1]
```diff
@@ -308,9 +308,12 @@ static port::Status InternalInit() {
 
   if (res == CUDA_SUCCESS) {
     return port::Status::OK();
+  } else if (res == CUDA_ERROR_SHARED_OBJECT_INIT_FAILED) {
+    LOG(WARNING) << "failed call to cuInit: " << ToString(res);
```
Comment (bhack) [inline tensorflow/stream_executor/cuda/cuda_driver.cc:312]: @chisgg Just to be clear `ToString(res)` as it was designed will be always classify also this case as `Unkown Error` for the LOG string but we really know that we are in `cudaErrorSharedObjectInitFailed = 303 This indicates that initialization of a shared object failed. `

**Pull request merged**